### PR TITLE
Enabling presence

### LINKED ISSUE #6: re-introduce presence

### PR DESCRIPTION
**Pull request by bhaviksingh**
# What Changed
Added server/.env.local, which will be preferred when running server 
Updated presence code in server to work
Sent presence and showed num online in client 

# Context
We had a half working presence system previously. I re-activated it, by updating the client to actually send the presence heartbeat, and the server to correctly process heartbeats and update numOnline in room.

In doing so, I realized that it's frustrating for the server to be working with the PROD firebase database. So I added a .env.local file to the server, and had the server read from it if available. This should be safe in prod which uses Fly secrets instead of .env files. 

# Validation

Tried it out with the local server, where everything seems to work. Once merged, will deploy fly. 

https://github.com/user-attachments/assets/58b73f9d-408a-467f-90d4-653ef180f37e

# Relevant issues
Closes #6

### CODE REVIEW

File: server/src/firestore-api.ts [hunk 1]
```diff
@@ -112,14 +113,18 @@ export const resetMuxFirestoreRelationship = async (roomID: string) => {
 /**  */
 export async function managePresenceInDB(allRoomNames: string[]) {
   const presenceRef = firestore.collection("presence");
-  
   const currentlyOnline = await Promise.all(allRoomNames.map(async (streamName) => {
-    let q = presenceRef.where("room_id", "==", streamName).where("timestamp", ">=", Timestamp.fromMillis(Date.now() - (1.2 * PRESENCE_LENGTH)));
+    //let q = presenceRef.where("room_id", "==", streamName).where("timestamp", ">=", Timestamp.fromMillis(Date.now() - (1.2 * PRESENCE_LENGTH)));
```
Comment (kyeah) [inline server/src/firestore-api.ts:117]: maybe we can remove these rogue commented snippets before merging 👍

File: client/src/components/room/footer.tsx [hunk 2]
```diff
@@ -21,8 +21,10 @@ const Footer: React.FC = () => {
         className="uiLayer horizontal-stack:s-2 padded:s-1 "
         style={{ position: "absolute", bottom: 0, right: 0 }}
       >
+        <NumOnline />
```
Comment (kyeah) [inline client/src/components/room/footer.tsx:24]: Note — I noticed `in room` still showed up on the homepage. We should either show...

- The total number of people on the site at the moment (in all rooms)
- The number of people on the homepage
- Nothing (hide this component)

**Review by kyeah — APPROVED** [review]
Works for me! There is a delay of up to ~5 seconds because of the presenceProcessor schedule — I suspect we can probably be smarter about that so it gets recalculated immediately when people enter or leave the room, but we can look into additional improvements in the future.

File: server/src/firestore-api.ts (hunk 1 above)
Comment (bhaviksingh) [reply server/src/firestore-api.ts]: Done!

File: client/src/components/room/footer.tsx (hunk 2 above)
Comment (bhaviksingh) [reply client/src/components/room/footer.tsx]: I updated this to show the total number of people on the site in all rooms.
It's happening primarily on the server, which is now writing to a new collection: stats/num_online.
Perhaps that will be useful for other things later too. The client then reads from this.

**Comment by bhaviksingh** [comment]
Thanks for all the comments, really really helpful. I made some changes to address your comments and made it snappier, so that the server updates presence every time something changes in presence directory (rather than on a schedule). Let me know what you think!

File: client/src/lib/firestore/stats.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+
+export function syncTotalOnline(callback: (stats: number) => void) {
+  const totalOnline = doc(statsCollection(), "total_online");
+
+  
+  const unsubscribe = onSnapshot(totalOnline, (docSnapshot) => {
+    if (docSnapshot.exists()) {
+      const data = docSnapshot.data() as TotalOnline;
+      console.log("ASDADASDADADASDAD", data);
```
Comment (kyeah) [inline client/src/lib/firestore/stats.ts:17]: 😂

**Review by kyeah — APPROVED** [review]
niceee, looks great and worked perfectly for me! 🙏 Looks like there's one build failure on Netlify due to this:

```
'numOnline' does not exist in type 'RoomLinkInfo'
```

**Pull request merged**